pkg/registry/batch/job/strategy.go

```diff
@@ -100,6 +101,9 @@ func (jobStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 	if !utilfeature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) {
 		job.Spec.PodFailurePolicy = nil
 	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.JobManagedBy) {
```

Comment:
noting for myself that validation prevents changing the managedBy field, so the PrepareForUpdate doesn't need to do anything to actively prevent someone from setting on an update

Reply:
I think it would be good to put such a comment into the PrepareForCreate/PrepareForUpdate code as well.

Reply:
Actually, I didn't ask for clarification since this was "noting to myself", but I think what prevents "setting on an update" is actually here: https://github.com/kubernetes/kubernetes/blob/3b728acd24c13eb948cd37a3108fa2aa681c0b15/pkg/apis/batch/validation/validation.go#L550, which prevents any type of mutation.

In my understanding this line is done to drop the field to avoid confusing clients that the field is set (and effective), while the feature is disabled.

@deads2k it would be great if you can elaborate more how this is also related to preventing setting on an update. I'm happy to transform the note into comments for future readers.

Reply:
Ah, got it, nice to learn something :) If the field was mutable we would need the code snippet to protect against a user adding the field when the feature-gate is off. 

Since the field is immutable we don't need that code. Together with the fact that we drop it on create (in PrepareForCreate) we protect the field being set when the feature gate is off.

What might be worth noting is that it is still be possible to end up with the field set and feature-gate off after cluster downgrade, or disabling the field in the KCM manifests, but both scenarios require admin / OnCall intervention.

Actually, I followed the pattern (which seems idiomatic now) for other immutable fields we have in Job: https://github.com/kubernetes/kubernetes/blob/9043ce05c125091c0cb5519206fd90d311abd8c8/pkg/apis/batch/validation/validation.go#L452-L454.

@atiratree let me know if you think think this still warrants a comment (after the explanations), and if so where / what type. Asking to confirm, because there are many layers at which this might be not obvious, and I don't want to overshoot with it.